src/java_bytecode/java_object_factory.cpp

```diff
@@ -151,7 +151,7 @@ class java_object_factoryt
 /// Generates code for allocating a dynamic object. This is used in
 /// allocate_object and for allocating strings in the library preprocessing.
 /// \param target_expr: expression to which the necessary memory will be
-///   allocated
+///   allocated, its type should be pointer to `allocate_type`
```

Comment:
Now that allocate_type is not a parameter, I'm not sure this comment is correct (or rather, it is trivially correct)